bundle/src/main/java/com/adobe/acs/commons/properties/impl/PropertyAggregatorServiceImpl.java

```diff
@@ -0,0 +1,273 @@
+     */
+    private void addPagePropertiesToMap(final Map<String,Object> targetPropertyMap, final Page page, final String prefix) {
+        ValueMap pageProperties = page.getProperties();
+        Set<Map.Entry<String, Object>> entries = pageProperties.entrySet();
+        for (Map.Entry<String, Object> entry : entries) {
+            if (notInMap(targetPropertyMap, entry.getKey())
+                    && isNotExcluded(entry.getKey())
+                    && isAllowedType(pageProperties.get(entry.getKey()))) {
+                targetPropertyMap.put(prefix(prefix, entry.getKey()), pageProperties.get(entry.getKey()));
+            }
+        }
+    }
+
+    /**
+     * Recursively traverse the properties of a page and it's parents until that parent is not a page or is null, which
+     * means it's the root, and add them to the given map.  The traversing will stop when the parent is a node that is
+     * not a page, such as a folder.  This is necessary to support things such as editable templates under `/conf`.
+     *
+     * @param targetPropertyMap the map that should be updated with the properties and their values
+     * @param page the page containing properties
+     */
+    private void addInheritedPagePropertiesToMap(final Map<String,Object> targetPropertyMap, final Page page) {
+
+        addPagePropertiesToMap(targetPropertyMap, page, INHERITED_PAGE_PROP_PREFIX);
+
+        Resource pageResource = page.adaptTo(Resource.class);
+        if (pageResource != null) {
+            Resource parentResource = pageResource.getParent();
+
+            if (parentResource != null && NameConstants.NT_PAGE.equalsIgnoreCase(parentResource.getResourceType())) {
+                addInheritedPagePropertiesToMap(targetPropertyMap, parentResource.adaptTo(Page.class));
+            }
+        }
+    }
+
+    /**
+     * Add additional data from custom Java beans that are defined in the configuration for this service to the given map.
+     * @param targetPropertyMap the map that should be updated with the properties and their values
+     * @param page the page for the context of the bean properties
+     */
+    private void addAdditionalDataToMap(final Map<String,Object> targetPropertyMap, final Page page) {
+        if (!additionalData.isEmpty()) {
+            for (Map.Entry<String, String> entry : additionalData.entrySet()) {
+                try {
+                    Resource contentResource = page.getContentResource();
+                    Class<?> modelClass = Class.forName(entry.getValue());
+                    Object model = contentResource.adaptTo(modelClass);
+                    if (model != null) {
+                        addBeanToMap(targetPropertyMap, entry.getKey(), model);
+                    }
+                } catch (ClassNotFoundException e) {
+                    log.error("Error mapping to class " + entry.getValue(), e);
+                } catch (Exception e) {
+                    log.error("Unknown error", e);
```

Comment:
Can you please provide more context to this log message (and the one above)? When debugging this it would be extremely useful to have the path of the page and the class name of the modelclass for which this exception was thrown.